tests/helper_functions.py

```diff
@@ -406,3 +409,33 @@ def decorator(f):
         return pytest.mark.allow_infer_pip_requirements_fallback(f) if condition else f
 
     return decorator
+
+
+def mock_method_chain(mock_obj, methods, return_value=None, side_effect=None):
+    """
+    Mock a chain of methods.
+
+    Examples
+    --------
+    >>> m = mock.MagicMock()
+    >>> mock_method_chain(m, ["a", "b"], 0)
+    >>> m.a().b()
+    0
+    >>> mock_method_chain(m, ["c.d", "e"], 1)
+    >>> m.c.d().e()
+    1
+    """
+    length = len(methods)
+    for idx, method in enumerate(methods):
+        mock_obj = functools.reduce(getattr, method.split("."), mock_obj)
+        if idx != length - 1:
+            mock_obj = mock_obj.return_value
+        else:
+            mock_obj.return_value = return_value
+            mock_obj.side_effect = side_effect
+
+
+@contextmanager
+def multi_context(*cms):
+    with ExitStack() as stack:
+        yield list(map(stack.enter_context, cms))
```

Comment:
@liangz1 Defined this function so we can write multi-line 

```python
    with multi_context(
        experiment_id_patch,
        databricks_notebook_patch,
        user_patch,
        source_name_patch,
        source_type_patch,
        source_version_patch,
        create_run_patch,
    ):
```

Reply:
Cool!